jerry-core/vm/vm.c

```diff
@@ -252,6 +253,23 @@ vm_run_eval (ecma_compiled_code_t *bytecode_data_p, /**< byte-code data */
   {
     this_binding = ecma_copy_value (JERRY_CONTEXT (vm_top_context_p)->this_binding);
     lex_env_p = JERRY_CONTEXT (vm_top_context_p)->lex_env_p;
+
+    while (chain_index != 0)
+    {
+      lex_env_p = ecma_get_lex_env_outer_reference (lex_env_p);
+
+      if (JERRY_UNLIKELY (lex_env_p == NULL))
+      {
+        lex_env_p = JERRY_CONTEXT (vm_top_context_p)->lex_env_p;
```

Comment:
Perhaps we could use the global lex_env instead, which is always the last. Something like:
```
if (JERRY_UNLIKELY (lex_env_p == jerry_get_global_lex_env())) {
  break;
}
```

The global lex env could be stored in a local variable to speed things up. We should enter to this loop if `chain_index > 0` which should be guarded with JERRY_UNLIKELY.